### PR TITLE
JBPM-6672 - Stunner - Add support for polylines

### PR DESCRIPTION
**Pull request by Josephblt**
Stunner - Add support for polylines

- Connectors now support polylines.
- Polyline segments now snap at straight angles.

JIRA:
https://issues.jboss.org/browse/JBPM-6672

This Pull Request depends on this:
Lienzo-core
https://github.com/kiegroup/lienzo-core/pull/6
Only merge after this other pull request gets merged.

@romartin 
@tiagodolphine 
@LuboTerifaj

### CODE REVIEW

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Hey @Josephblt , just a note for this and further PRs, it's really important to add in the PR description the dependencies with any other PRs. For example, the dependency to https://github.com/kiegroup/lienzo-core/pull/6 for this one.
This is to ensure it gets merged with the other dependent PRs as well, and nothing will get broken. 
So please, can you update the description of the PRs for the polylines and just add the links between them? Thanks a lot, that save us lot of time and potential issues when merging,

**Comment by Josephblt** [comment]
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
@romartin fixed the DMN module it had problems just like you said. But I still can't build the whole project because of another error that is not related to my work. It's a indexing test that fails:
IndexDrlLHSTypeExpressionFieldCombinedTest.testIndexDrlLHSTypeExpressionField
Other than that I believe everything else is fine.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-shapes/kie-wb-common-stunner-shapes-client/src/test/java/org/kie/workbench/common/stunner/shapes/client/view/OrthogonalPolyLineConnectorViewTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import org.kie.workbench.common.stunner.shapes.client.factory.OrthogonalPolyLineConnectorFactory;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+@RunWith(LienzoMockitoTestRunner.class)
+public class OrthogonalPolyLineConnectorViewTest {
+
+    @Test
+    public void testCreateLine() {
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-shapes/kie-wb-common-stunner-shapes-client/src/test/java/org/kie/workbench/common/stunner/shapes/client/view/OrthogonalPolyLineConnectorViewTest.java]: Also here, could we declare variables as `final` please?

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-shapes/kie-wb-common-stunner-shapes-client/src/test/java/org/kie/workbench/common/stunner/shapes/client/view/PolyLineConnectorViewTest.java [hunk 2]
```diff
@@ -22,17 +22,20 @@
...
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 
 @RunWith(LienzoMockitoTestRunner.class)
-public class ConnectorViewTest {
+public class PolyLineConnectorViewTest {
 
     @Test
     public void testCreateLine() {
```
Comment (jomarko) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-shapes/kie-wb-common-stunner-shapes-client/src/test/java/org/kie/workbench/common/stunner/shapes/client/view/PolyLineConnectorViewTest.java]: Last `final`, could we please use also here?

**Comment by hasys** [comment]
Full downstream build is not provided, some disk quota exceeded.
Jenkins execute full downstream build

**Comment by Josephblt** [comment]
Het @hasys. I'm glad you liked it. I think I can fix this bug. Will talk to Roger and assign this ticket to me.
@jomarko will make those changes as soon as I can.

**Comment by tiagodolphine** [comment]
Hey @hasys about the JBPM-6882 great, and thanks to opened it, actually we discussed this bug on the bendpoints PR, we need to solve that on Lienzo, IMO not related to this PR, but sure we need to solve that when it becomes the priority then just move to the sprint.

Thanks.

**Pull request merged**